nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -0,0 +1,534 @@
+            magnitude = self._get_bias_shift_magnitude(current_bias, updated_bias)
+
+            if magnitude < self.threshold:
+                nncf_logger.info(f'{node_name} bias would be changed. Magnitude: {magnitude}')
+                bias_port_id = self._backend_entity.get_bias_port_id(model, node)
+                target_point = self._backend_entity.target_point(TargetType.LAYER,
+                                                                 node.node_name,
+                                                                 bias_port_id)
+                bias_correction_command = self._backend_entity.bias_correction_command(target_point,
+                                                                                       updated_bias)
+                model_copy_subgraph = self._correct_bias(model_copy_subgraph, bias_correction_command)
+                model_copy = self._correct_bias(model_copy, bias_correction_command)
+                main_transformations_layout.register(bias_correction_command)
+            else:
+                nncf_logger.info(f'{node_name} bias skipped by threshold. Magnitude: {magnitude}')
+
+            self._collect_new_stats(nncf_graph, model_copy_subgraph, feed_dicts, subgraph_data)
+            self._remove_unnecessary_stats(node_name, subgraphs_data)
+        return main_model_transformer.transform(main_transformations_layout)
+
+    def _remove_fq_from_inputs(self, model: TModel) -> TModel:
+        """
+        This model removes the activation Fake Quantize nodes (or Quantize-Dequantize pairs) from the model.
+        It's needed for the further bias shift calculation that relates on quantized weights.
+
+        :param model: Backend-specific model.
+        :return: Backend-specific model without activation Fake Quantize nodes (or Quantize-Dequantize pairs).
+        """
+        transformation_layout = TransformationLayout()
+        skip_types = []
+        nncf_graph = NNCFGraphFactory.create(model)
+
+        model_transformer = self._backend_entity.model_transformer(model)
+        for skip_type in self._backend_entity.quantizer_types:
+            skip_types.extend(skip_type.op_names)
+
+        seen_nodes = []
+        nodes_queue = deque(nncf_graph.get_input_nodes())
+        while nodes_queue:
+            current_node = nodes_queue.popleft()
+            current_node_name = current_node.node_name
+
+            if current_node_name in seen_nodes:
+                continue
+
+            seen_nodes.append(current_node_name)
+            if current_node.node_type in skip_types:
+                target_point = self._backend_entity.target_point(
+                    TargetType.LAYER, current_node_name, 0)
+                command = self._backend_entity.node_removing_command(target_point)
+                transformation_layout.register(command)
+            nodes_queue.extend(nncf_graph.get_next_nodes(current_node))
+
+        return model_transformer.transform(transformation_layout)
```

Comment:
To be honest, I'm waiting that the model transformer returns a new model instance. I mean it doesn't apply operations in-place.

Reply:
Not now, we'll cover that while adding other backends.